app/lib/three_scale/sidekiq_redis_config.rb

```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+module ThreeScale
+  class SidekiqRedisConfig < RedisConfig
+    def initialize(redis_config = {})
+      redis_config.delete(:namespace)
+      super redis_config
+    end
+  end
+end
```

Comment:
We allow customers to use Redis namespaces, but Sidekiq 7 removed support for that. So we need now two Redis configuration parsers, one accepting namespaces for Porta and one not accepting them for Sidekiq.

I would advise customers against using namespaces anymore, because I don't know the effects of using namespaces only for porta but not for Sidekiq. For instance, some features like the Redis lock for billing jobs would use namespaces but the jobs would be stored and consumed using Redis DB ids.

If a customer is using Sidekiq with namespaces now, they will have to migrate to Redis DB ids.

Reply:
Do we have to notify customers about this? It sounds like a breaking change.

So if `redis_config.delete(:namespace)` is not called, Sidekiq will throw an error?

Reply:
I guess that the use case he would be that the customers have a single instance of Redis used by multiple applications, one of them being 3scale. So they set `REDIS_NAMESPACE=threescale`, and all 3scale-related keys will be seamlessly prefixed by `threescale:`, which would kind of "guarantee" that there will be no clash with other keys in the same Redis database.

Actually, I'm not sure where exactly Redis is used by porta outside of Sidekiq, I've found just a couple of places:
- the locking mechanism now used by billing
- web hook failures
- there is some "rate limiting" module, but I doubt it's actually used

So, I think most keys stored in Redis by porta are Sidekiq keys.

As such, it doesn't make sense to keep using Redis namespaces for porta, but not for Sidekiq, as it would for sure be a potential conflict.

I'd say we need to deprecate using Redis namespaces and force all customers use a Redis database dedicated to 3scale porta uniquely.
Maybe we can take advantage of 2.14 to sent a "note" to the customers (while not actually changing anything in the code - although some deprecation messages in the logs would be nice too).

Reply:
Also, testing the `REDIS_NAMESPACE` locally now, I noticed some deprecation warnings:
```
Passing 'exists?' command to redis as is; blind passthrough has been deprecated and will be removed in redis-namespace 2.0 (at /home/dmayorov/.asdf/installs/ruby/2.7.6/lib/ruby/gems/2.7.0/gems/sidekiq-6.4.2/lib/sidekiq/launcher.rb:167:in `block (2 levels) in ❤')
```
this appears in the logs every couple of minutes.
Related: https://github.com/resque/redis-namespace/issues/172

```
Passing 'script' command to redis as is; administrative commands cannot be effectively namespaced and should be called on the redis connection directly; passthrough has been deprecated and will be removed in redis-namespace 2.0 (at /home/dmayorov/Projects/3scale/porta/app/lib/system/redis_pool.rb:26:in `public_send')
```

This appeared on a manual action.

Related: https://github.com/sidekiq/sidekiq/pull/5066

Look like it would be best to get rid of Redis namespaces :fire:

Reply:
Good idea, to deprecate in 2.14. We need to log a warning as well.

Another user of Redis is activerecord-uniqueness but it uses it the same as the billing locking - through `redlock`.

Also from Porta we connect to the backend Redis. But perhaps that is not affected. btw I think that both locking mechanisms use the backend redis, not the sidekiq one. We may want to change that or if easier, leave as is.

Reply:
We will have to make sure jobs are drawn completely before upgrade though. Otherwise we may have jobs in a namespace that will never be run. So upgrade mechanism will need to be figured out.

Reply:
Nooo, there is no way backend redis is used for locking! Backend redis should be used by System as read-only.

Reply:
I think `System.redis` is backend. I saw something like this a couple of weeks ago. Have to double check but we may need to update locking not only for billing but also for activerecord-uniqueness

Reply:
I'd say the only place in the code where the backend redis configuration is read is here:

https://github.com/3scale/porta/blob/b5685ab58f27b9ac60e376a1d3af1ee94c3efc78/app/lib/backend/storage.rb#L7

All other interactions with backend are done through the `BackendClient::Connection` class, using the API, not Redis. 

WRT `System.redis`, it's set at:

https://github.com/3scale/porta/blob/89a91607b8387892967a0b9315cca9e0d36c99e8/config/application.rb#L315

And used from:

- Redlock, which is fine.
- `WebHookFailures`:
  https://github.com/3scale/porta/blob/25fa9bc53bdd3eeb8f2a9c8e882f771c85c3c3cb/app/models/web_hook_failures.rb#L105
- `ActionLimiter`:
  https://github.com/3scale/porta/blob/01f16b96f0a89aea0d5b3278397feedeca6fdc6d/app/services/action_limiter.rb#L26

I'd say webhooks are fine, but action limiter seems to be related to backend, @akostadinov @mayorova can you confirm?